server/src/providers/blockchain.ts

```diff
@@ -376,8 +377,7 @@ export class blockchain {
 
     if (allAddresses.length) {
       // there could be duplicates, so need to get unique values
-      // new Set([...]) -> will return unique values from received array
-      const uniqueAddresses = [...new Set(allAddresses)]
+      const uniqueAddresses = uniq(allAddresses)
```

Comment:
isn't uniqBy(uniqueAddresses, toLower) working ?

Reply:
```
uniqBy(allAddresses, toLower)
```
It just checks the values being lower case applied but return the original value.

Reply:
@yaroslav-fedyshyn-nordwhale yep i would have exactly that behavior

Reply:
done